### PR TITLE
9 More Traits

### PR DESCRIPTION
**Pull request by VMSolidus**
# Description

This "ZERO C#" Pr adds 10 additional traits to the game, which only make use of pre-existing components, along with some of the new TraitSystem functionality. These are the following new traits:

- **Photophobia**: Functions as per Vulpkanin Light Sensitivity
- **Clumsy**: Functions as per Clown clumsiness
- **Small**: Functions as per Felinid PseudoItem, requires a character be Felinid sized
- **Temperature Tolerance**: Functions as per Vulpkanin low temperature resistance
- **Spinarette**: Functions as per the same ability from Arachnids.
- **Talons**: Replaces your melee natural attacks with Piercing talons
- **Claws**: Replaces your melee natural attacks with Slashing claws
- **Striking Calluses**: [Human only, requires Martial Artist or Boxer job], increases your base unarmed attack to 6 damage, from 5.
- **Natural Weapon Removal**: For species with melee attacks other than blunt, replaces Talons or Claws with Fist. 

## TODO:

- [ ] Adjust the points costs/additions for these traits to ensure balance. Although I can't imagine most of them being changed. I do still wish for a decent number of traits to be within the 1 to 3 point range. 

<details><summary><h1>Media</h1></summary>
<p>

image

image

</p>
</details>

# Changelog

:cl:
- add: 10 new Traits have been added to the game. Photophobia, Clumsy, Small, Temperature Tolerance, Claws, Talons, Natural Weapon Removal, Striking Calluses, and Spinarette.

### CODE REVIEW

File: Resources/Prototypes/Traits/physical.yml [hunk 1]
```diff
@@ -231,3 +231,127 @@
...
+        - Small
+    - !type:CharacterHeightRequirement
+      max: 133
+    - !type:CharacterWidthRequirement
+      max: 29
+  componentRemovals:
+    - PseudoItem
+  components:
+    - type: Item
+      size: Normal
```
Comment (FoxxoTrystan) [inline Resources/Prototypes/Traits/physical.yml:277]: ARE YOU SURE ABOUT THIS?

File: Resources/Prototypes/Traits/physical.yml [hunk 2]
```diff
@@ -231,3 +231,127 @@
...
+    - MeleeWeapon
+  components:
+    - type: MeleeWeapon
+      soundHit:
+        collection: AlienClaw
+      angle: 30
+      animation: WeaponArcClaw
+      damage:
+        types:
+          Slash: 5
```
Comment (FoxxoTrystan) [inline Resources/Prototypes/Traits/physical.yml:340]: This trait also feels wired, i feel species melee attack should be kept to the species.

File: Resources/Prototypes/Traits/physical.yml [hunk 3]
```diff
@@ -231,3 +231,127 @@
...
+      inverted: true
+      species:
+        - Arachnid
+        - Arachne
+  components:
+    - type: Sericulture
+      action: ActionSericulture
+      productionLength: 2
+      entityProduced: MaterialWebSilk1
+      hungerCost: 4
```
Comment (FoxxoTrystan) [inline Resources/Prototypes/Traits/physical.yml:357]: Unless we make a "Custom Specie" system where the player make his own specie in-game, i feel this should not be selectable and leave it for those species.

File: Resources/Prototypes/Traits/physical.yml [hunk 4]
```diff
@@ -231,3 +231,127 @@
...
+  componentRemovals:
+    - MeleeWeapon
+  components:
+    - type: MeleeWeapon
+      soundHit:
+        collection: AlienClaw
+      animation: WeaponArcClaw
+      damage:
+        types:
+          Piercing: 5
```
Comment (FoxxoTrystan) [inline Resources/Prototypes/Traits/physical.yml:314]: Piercing damange is... dangerous.

Andd this trait feels wired to give it to everyone.

File: Resources/Prototypes/Traits/physical.yml (hunk 1 above)
Comment (FoxxoTrystan) [reply Resources/Prototypes/Traits/physical.yml:277]: This seem somehow OP and... really annoying to just grab peoples.
I mean i seen it in high roleplays but with toggles.

Even still, this coulld make a ton of isssues regarding antags/combat.

**Review by FoxxoTrystan — COMMENTED** [review]
not approving but not request changing as those feels more like its for a "custom specie" slot then the rest.

In all honest some trait should be removed and kept for those species but, huh maybe that is just my personnal opinion.

But personnaly i would not give the claws, talon, spider trait to random species unless making a custom specie system.

File: Resources/Prototypes/Traits/physical.yml (hunk 4 above)
Comment (VMSolidus) [reply Resources/Prototypes/Traits/physical.yml:314]: Piercing damage is literally the same thing as the two other brute damage types, having exactly the same effects as the other two. You're straight up wrong.
Infact it's actually the least dangerous of all damage types, due to Piercing armor being the most common armor in the game.

File: Resources/Prototypes/Traits/physical.yml (hunk 3 above)
Comment (ShatteredSwords) [reply Resources/Prototypes/Traits/physical.yml:357]: > Unless we make a "Custom Specie" system where the player make his own specie in-game, i feel this should not be selectable and leave it for those species.

I do agree, people spinning webs is uh, kinda weird. Best you should do is maybe make it only available for moths? I know some type of moths make silk.

**Comment by VMSolidus** [comment]
> not approving but not request changing as those feels more like its for a "custom specie" slot then the rest.
> 
> In all honest some trait should be removed and kept for those species but, huh maybe that is just my personnal opinion.
> 
> But personnaly i would not give the claws, talon, spider trait to random species unless making a custom specie system.

This is the custom species system. Buy a shitton of "Mutation" traits, and DIY. You're welcome.

**Comment by FoxxoTrystan** [comment]
"Tiny"

> > not approving but not request changing as those feels more like its for a "custom specie" slot then the rest.
> > In all honest some trait should be removed and kept for those species but, huh maybe that is just my personnal opinion.
> > But personnaly i would not give the claws, talon, spider trait to random species unless making a custom specie system.
> 
> This is the custom species system. Buy a shitton of "Mutation" traits, and DIY. You're welcome.

Ehh... personnaly for custom species system i would like to give servers a CVar to see if they want that enable or at least... hm...
Well fine i supose.

File: Resources/Prototypes/Traits/physical.yml (hunk 1 above)
Comment (VMSolidus) [reply Resources/Prototypes/Traits/physical.yml:277]: Yes.

**Comment by FoxxoTrystan** [comment]
Personnally i feel this would cause a mess.
And its feels a bit detached.

But i wana get others peoples opinion on it first before i approve it.

File: Resources/Prototypes/Traits/physical.yml (hunk 1 above)
Comment (VMSolidus) [reply Resources/Prototypes/Traits/physical.yml:277]: Counterpoint, you can put people in the deep fryer who take this trait.
You can also shove them into your pocket. Open an airlock. Then Ctrl-Q them into space.

**Comment by OldDanceJacket** [comment]
> Personnally i feel this would cause a mess. And its feels a bit detached.
> 
> But i wana get others peoples opinion on it first before i approve it.

Tagging in to review, I agree with the thoughts shared here. Turning players into Items has a lot of actual implications I don't like and I'd just like that one day you can put anyone in the deep-fryer regardless(I know there's more things possible but using that example). Small is 'okay' but I genuinely don't like the range of height and weight the sliders can currently go in the first place and have had a few other folks speak up about how thin and short you can make folks is a bit much.

Realistically I'd say get rid of Tiny, Talons, Claws, and Spinarette.

This is genetic weirdness that shouldn't be round start and is really against a lot of the vibes of SpaceStation.

The other traits are fine, Temperature Tolerance is a real thing I actually met a guy who would work in the walk in freezer shirtless just moving boxes from the truck.

**Comment by FoxxoTrystan** [comment]
Solidus was drunk when doing this PR.

**Review by DEATHB4DEFEAT — APPROVED** [review]

File: Resources/Prototypes/Traits/physical.yml [hunk 5]
```diff
@@ -231,3 +231,78 @@
...
+      max: 150
+    - !type:CharacterWidthRequirement
+      max: 32
+  components:
+    - type: PseudoItem
+      storedOffset: 0,17
+      shape:
+        - 0,0,1,4
+        - 0,2,3,4
+        - 4,0,5,4
```
[the remark this answers is not in the document]
Comment (Mnemotechnician) [reply Resources/Prototypes/Traits/physical.yml:258]: (X0, Y0, X1, Y1) bounding boxes comprising the item shape

**Comment by VMSolidus** [comment]
> > Personnally i feel this would cause a mess. And its feels a bit detached.
> > But i wana get others peoples opinion on it first before i approve it.
> 
> Tagging in to review, I agree with the thoughts shared here. Turning players into Items has a lot of actual implications I don't like and I'd just like that one day you can put anyone in the deep-fryer regardless(I know there's more things possible but using that example). Small is 'okay' but I genuinely don't like the range of height and weight the sliders can currently go in the first place and have had a few other folks speak up about how thin and short you can make folks is a bit much.
> 
> Realistically I'd say get rid of Tiny, Talons, Claws, and Spinarette.
> 
> This is genetic weirdness that shouldn't be round start and is really against a lot of the vibes of SpaceStation.
> 
> The other traits are fine, Temperature Tolerance is a real thing I actually met a guy who would work in the walk in freezer shirtless just moving boxes from the truck.

"This is genetic weirdness that shouldn't be round start and is really against a lot of the vibes of SpaceStation."
"Doesn't go with the vibes of SS13"
Okay. Absolutely not. In the case of bionic spinarette, this is literally a trait lifted directly 1:1 from SS13. You absolutely can be wrong about some things, and this is one of those. Also NEITHER of the small or tiny traits have anything at all to do with modifying your height sliders, they merely ask for your height slider bounds as a requirement. Which, I should remind you, I intentionally selected 1:1 X and Y scale limitations specifically so that you aren't being a "Lankly" character to minmax with the trait.

Hell I'm half tempted to put back claws and talons, because those are also shit you can just fucking do in SS13. 
image

Behold, a whole goddamn bionics menu. I would fucking love to have something like this one day, and in that time, remove the traits and move them to the bionics menu.

**Comment by VMSolidus** [comment]
Right, so I'm surprised that Drunk Solidus can do YML just fine, but his writing is dogshit. So I'm not only doubling down on this PR, but I'm also substantially increasing the quality and depth of the trait descriptions.

image

**Comment by OldDanceJacket** [comment]
> > > Personnally i feel this would cause a mess. And its feels a bit detached.
> > > But i wana get others peoples opinion on it first before i approve it.
> > 
> > 
> > Tagging in to review, I agree with the thoughts shared here. Turning players into Items has a lot of actual implications I don't like and I'd just like that one day you can put anyone in the deep-fryer regardless(I know there's more things possible but using that example). Small is 'okay' but I genuinely don't like the range of height and weight the sliders can currently go in the first place and have had a few other folks speak up about how thin and short you can make folks is a bit much.
> > Realistically I'd say get rid of Tiny, Talons, Claws, and Spinarette.
> > This is genetic weirdness that shouldn't be round start and is really against a lot of the vibes of SpaceStation.
> > The other traits are fine, Temperature Tolerance is a real thing I actually met a guy who would work in the walk in freezer shirtless just moving boxes from the truck.
> 
> "This is genetic weirdness that shouldn't be round start and is really against a lot of the vibes of SpaceStation." "Doesn't go with the vibes of SS13" Okay. Absolutely not. In the case of bionic spinarette, this is literally a trait lifted directly 1:1 from SS13. You absolutely can be wrong about some things, and this is one of those. Also NEITHER of the small or tiny traits have anything at all to do with modifying your height sliders, they merely ask for your height slider bounds as a requirement. Which, I should remind you, I intentionally selected 1:1 X and Y scale limitations specifically so that you aren't being a "Lankly" character to minmax with the trait.
> 
> Hell I'm half tempted to put back claws and talons, because those are also shit you can just fucking do in SS13. image
> 
> Behold, a whole goddamn bionics menu. I would fucking love to have something like this one day, and in that time, remove the traits and move them to the bionics menu.

It doesn't fit the vibes, doesn't matter if some forks put it in or not, I'm giving my opinion.

Secondly the height/weights were a minor point, I don't care about the minmax attempts of it(small and weird as that is), I care that the Tiny trait and the implications it brings as well as the round start gameplay it gives to the players is inherently just toxic and I dislike it a bunch. 
I told you in DMs when you were drunk that beyond all of the other additions you want to make this one is the one I have the most problems with and frankly it's just dumb. There doesn't need to be a trait like this in the game.

File: Resources/Prototypes/Traits/physical.yml [hunk 6]
```diff
@@ -231,3 +231,173 @@
...
+  id: Claws
+  category: Physical
+  points: -1
+  requirements:
+    - !type:CharacterSpeciesRequirement
+      inverted: true
+      species:
+        - Felinid # Felinids already have cat claws.
+        - Reptilian # Reptilians also have cat claws.
+        # - Vulpkanin # Vulpkanin have "Blunt" claws. One could argue this trait "Sharpens" their claws.
```
Comment (FoxxoTrystan) [inline Resources/Prototypes/Traits/physical.yml:308]: they do? time to fix it later and then cchange it here.

**Review by FoxxoTrystan — APPROVED** [review]
I dont hate this

**Pull request merged**